### ISSUE TITLE
Get currency by code string

### ISSUE THREAD

**Issue by jlalvarez18** [danthorpe/Money#29]
Is there any way we can get a Currency by using the code?

```
let c = Currency(code: "USD")
```

It would be nice construct a Currency object by the code itself. This helps in situations where you get the currency code from a backend.

**Comment by danthorpe** [comment]
Hi, thanks for the feedback!

At the moment with v1.4 while there is a constructor to sort of do this, but it isn't made public.

I have thought about enabling this kind of functionality, just haven't had time to really think clearly about how it should be done, with the focus on safety. Taking your example code...

```swift
if let currency = Currency(code: "USD") {
  // it would need to be a fail-able initializer
}
```
but, now how should the `currency` value be used? Given that `_Money` is generic over `CurrencyType` - there is no need to instantiate a `CurrencyType` from the point of view of Money.

So, because of the way I've made `CurrencyType` and `MoneyType` constructing values based on a string is a little tricky - because at compile time you don't know what the eventual type is. So, for example, writing a function like this:

```swift
if let dollars = Currency.createMoneyWithCode("USD", minorUnits: 100) {
   XCTAssertEquals(dollars, USD(1))
}
``` 
will not compile, because the compiler will not know what the `Currency` of `dollars` (which will be  a `MoneyType`) is. This is the case with functions which return generics. And typically you'd need to do something like...

```swift
if let dollars: USD = Currency.createMoneyWithCode("USD", minorUnits: 100) {
   // this would work - but defeats the purpose
}
```

which obviously defeats the purpose.

So... the way forward, is to either possibly re-architect the currencies and maybe use a `RawRepresentable` enum in there somewhere. Or, create an `AnyMoneyType` type erased value for this use case.

So, it's now bumped to the top of my list of things to do - keep an eye on this issue.

**Comment by wess** [comment]
👍 

Stripe, for instance, sends back currency codes as strings, like "gdp" so this would be huge.

**Comment by danthorpe** [comment]
Yeah, unfortunately, it's not quite as simple as you might think to do this with the framework as it currently stands. I've got a few ideas, but just haven't quite found the time to get it done. Hopefully will get to it soon!

**Comment by Sega-Zero** [comment]
Any news on this?

**Comment by danthorpe** [comment]
No work has been done on this - focus has been elsewhere as of late. Will hopefully re-visit _Money_ later on this year.

**Comment by kenyth** [comment]
Also looking forward to this enhancement.

**Comment by dongyuan** [comment]
+1, It would be great if we can have a generic `AnyMoneyType`.
for example, I am using `Moya` lib and created a struct to reflect the object returned by the server, but I cannot do it now.

**Comment by AtharvaVaidya** [comment]
+1 would like this feature implemented. I need to keep track of the currency being used.

**Comment by Firefox2005** [comment]
+1 Dynamic currencies initialized with the ISO-code would be great.

**Comment by Sega-Zero** [comment]
Made [my implementation](https://github.com/Sega-Zero/swift-iso-4217), maybe it could help?

**Comment by RamblinWreck77** [comment]
Would like this feature as well. Not sure how I can use this without this feature, since I can get an input in ANY currency type. I can't have a several-hundred currency switch. I feel like this feature should be a high priority one.
